### PR TITLE
fix(lambda): invalid Version object created from Version.fromVersionArn

### PR DESCRIPTION
**Pull request by GavinZZ**
### Issue # (if applicable)

Closes https://github.com/aws/aws-cdk/issues/29813

### Reason for this change

The current `lambda.Version.fromVersionArn` will incorrectly create the lambda function arn with the version qualifier. This is incorrect behaviour and cause CFN deployment failures.

### Description of changes

If the version arn is an unresolved token, use intrinsics to join the first 7 components. Otherwise, split the array and join the first 7 components directly.

### Description of how you validated changes

All new and existing tests should pass. New integration tests added with assertions.

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by paulhcsun — APPROVED** [review]
Changes here look good to me and should not run into the same issue as the previous fix. Will approve for now but wait for a second opinion before we merge.

**Review by moelasmar — APPROVED** [review]

**Pull request merged**
